### PR TITLE
ApiResponse: allow statusCode for response code

### PR DESCRIPTION
**Pull request by amritashan**

### CODE REVIEW

**Comment by Rush** [comment]
Thanks for this PR. Can you add a test for this new functionality in test.js?

File: src/index.ts [hunk 1]
```diff
@@ -165,11 +165,11 @@ function toMiddleware(this: ExpressApiRouter,
         }),
         catchError((err: Error) => {
           return resolve(formatError(err)).pipe(map((jsonError) => {
-            return new ApiResponse(jsonError || internalServerError, 500);
+            return new ApiResponse(jsonError || internalServerError, (jsonError as any).statusCode || 500);
```
Comment (Rush) [inline src/index.ts]: jsonError may be undefined it seems. reading `.statusCode` from undefined would fail

File: src/index.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ function toMiddleware(this: ExpressApiRouter,
     const formatError = (err: Error): AsyncApiResult => {
       if (this.errorFormatter) {
         return defer(() => resolve(this.errorFormatter(err, req, res)))
-          .pipe(map((formatted) => new ApiResponse(formatted, 500)));
+          .pipe(map((formatted) => new ApiResponse(formatted, (err as any).statusCode || 500)));
```
Comment (Rush) [inline src/index.ts]: Please add a test case

File: src/index.ts [hunk 3]
```diff
@@ -165,11 +165,11 @@ function toMiddleware(this: ExpressApiRouter,
         }),
         catchError((err: Error) => {
           return resolve(formatError(err)).pipe(map((jsonError) => {
-            return new ApiResponse(jsonError || internalServerError, 500);
+            return new ApiResponse(jsonError || internalServerError, (jsonError as any).statusCode || 500);
           }));
         }),
         catchError((err: Error) => {
-          return of(new ApiResponse(internalServerError, 500));
+          return of(new ApiResponse(internalServerError, (err as any).statusCode || 500));
```
Comment (Rush) [inline src/index.ts]: Please add a test case

**Comment by amritashan** [comment]
Addressed review comments and added tests.

**Pull request merged**